app/representers/concerns/embeds.rb

```diff
@@ -63,14 +63,14 @@ def embed(name, options={})
           no_curies: true,
           item_class: options.delete(:item_class),
           url: options.delete(:url),
-          item_decorator: options.delete(:item_decorator),
-          per: options.delete(:per) || Kaminari.config.default_per_page
+          item_decorator: options.delete(:item_decorator)
         }
+        getter_per = options.delete(:per) || Kaminari.config.default_per_page
+        getter_name = options.delete(:get) || name
         options[:getter] ||= ->(*) do
           # set # per page based on default, option value integer, or special :all
-          per = opts.delete(:per)
-          per = self.send(name).count if per == :all
-          PagedCollection.new(self.send(name).page(1).per(per), nil, opts.merge({parent: self}))
+          per = send(getter_name).count if getter_per == :all
+          PagedCollection.new(send(getter_name).page(1).per(per), nil, opts.merge(parent: self))
```

Comment:
I believe you, probably an oversight on my part, thanks for fixing

On Fri, Apr 8, 2016 at 2:12 PM, cavis <notifications@github.com> wrote:

> In app/representers/concerns/embeds.rb
> <https://github.com/PRX/cms.prx.org/pull/133#discussion_r59066882>:
>
> >          options[:getter] ||= ->(*) do
> >            # set # per page based on default, option value integer, or special :all
> > -          per = opts.delete(:per)
> > -          per = self.send(name).count if per == :all
> > -          PagedCollection.new(self.send(name).page(1).per(per), nil, opts.merge({parent: self}))
> > +          per = send(getter_name).count if getter_per == :all
> > +          PagedCollection.new(send(getter_name).page(1).per(per), nil, opts.merge(parent: self))
>
> Will get rid of the :get option. But do you agree that this shouldn't be
> opts.delete(:per)-ing inside the getter? I think that would work normally
> on the first call, but on the second call you would get .page(1).per(nil).
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/PRX/cms.prx.org/pull/133/files/b2914671cd5040d450a5fb0d6100234f0d594cb1#r59066882>
>